### ISSUE TITLE
SIM117 false alert mixing `async with` and `with` statements

### ISSUE THREAD

**Issue by dmoklaf** [astral-sh/ruff#11834]
[SIM117](https://docs.astral.sh/ruff/rules/multiple-with-statements/) is triggered by this code:
```
with localhost.keep_awake():
        async with locking.lock("xxx"):
                ...
```
This is a false alert as it is impossible to merge a `with` statement and an `async with` statement together.

Keywords: `SIM117`
Command: `ruff check .`
Ruff settings: none
Ruff version: `ruff 0.4.8`

**Comment by AlexWaygood** [comment]
Hi, thanks for the report! Unfortunately I can't reproduce this locally or in the playground: https://play.ruff.rs/7d1da502-34d7-4833-a3e1-7dacdc65d03a. Would you be able to provide a [minimal, reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)? :)

**Comment by dmoklaf** [comment]
Of course. My current code behind these calls is super heavy. Will extract a minimal reproducible example tomorrow (wednesday).

**Comment by dmoklaf** [comment]
This is a false alert ! There was below the first 2 lines extracted above a third line with another `async with` context. So the `SIM117` alert was legitimate. Sorry for the noise.

**Comment by AlexWaygood** [comment]
Great to hear @dmoklaf.

> Sorry for the noise.

No worries -- thanks for taking the time to write up a bug report!
